tests/unit/device_comms/test_mqtt_device_comms.py

```diff
@@ -0,0 +1,124 @@
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import os
+import unittest
+
+from liota.device_comms.mqtt_device_comms import MqttDeviceComms
+from liota.lib.transports.mqtt import Mqtt
+
+config = {}
+execfile(os.path.dirname(os.path.abspath(__file__)) + '/conf/testMqttDeviceCommsProp.conf', config)
+
+
+# Monkey patching disconnect method
+def mock_disconnect(self, *args, **kwargs):
+    pass
+
+
+# Monkey patching connect method
+def mock_connect_soc(self, *args, **kwargs):
+    pass
+
+
+# Monkey patching publish method
+def mock_publish(self, *args, **kwargs):
+    pass
+
+
+# Monkey patching subscribe method
+def mock_subscribe(self, *args, **kwargs):
+    pass
+
+
+class MqttDeviceCommsTest(unittest.TestCase):
+    """
+    Unit test cases for MQTT DeviceComms
+    """
+
+    def setUp(self):
+        """
+        Setup all required parameters for MQTT device communication tests
+        :return:
+        """
+        # Monkey patched MQTT connect method
+        Mqtt.connect_soc = mock_connect_soc
+
+        self.url = config['BrokerIP']
+        self.port = config['BrokerPort']
+        self.clean_session = True
+        self.publish_topic = "publish-topic"
+        self.subscribe_topic = "subscribe-topic"
+        self.publish_message = "test-message"
+
+        # Instantiate client for DCC communication
+        self.client = MqttDeviceComms(url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def test_init(self):
+        """
+        Test MqttDeviceComms initialisation
+        :return:
+        """
+        self.assertIsNotNone(self.client)
+
+    def test_disconnect(self):
+        """
+        Test MqttDeviceComms _disconnect method
+        :return:
+        """
+        # Monkey patched MQTT disconnect method
+        Mqtt.disconnect = mock_disconnect
+
+        self.assertIsNone(self.client._disconnect())
+
+    def test_publish(self):
+        """
+        Test MqttDeviceComms publish method
+        :return:
+        """
+        # Monkey patched MQTT publish method
+        Mqtt.publish = mock_publish
+
+        self.assertIsNone(self.client.publish(topic=self.publish_topic, message=self.publish_message, qos=1))
+
+    def test_subscribe(self):
+        """
+        Tests MqttDeviceComms subscribe method
+        :return:
+        """
+        # Monkey patched MQTT subscribe method
+        Mqtt.subscribe = mock_subscribe
+
+        self.assertIsNone(self.client.subscribe(topic=self.subscribe_topic, qos=1, callback=None))
+
+if __name__ == '__main__':
+    unittest.main(verbosity=1)
```

Comment:
Ensure there is EOF.

Also, follow all review comments from dcc_comms to device_comms test cases also.